substrate/frame/broker/src/tests.rs

```diff
@@ -891,6 +891,164 @@ fn short_leases_are_cleaned() {
 	});
 }
 
+#[test]
+fn leases_can_be_renewed() {
+	TestExt::new().endow(1, 1000).execute_with(|| {
+		// Timeslice period is 2.
+		//
+		// Sale 1 starts at block 7, Sale 2 starts at 13.
+
+		// Add a core.
+		assert_ok!(Broker::request_core_count(RuntimeOrigin::root(), 1));
+
+		// Set lease to expire in sale 1 and start sales.
+		assert_ok!(Broker::do_set_lease(2001, 9));
+		assert_eq!(Leases::<Test>::get().len(), 1);
+		// Start the sales but don't offer any more cores.
+		assert_ok!(Broker::do_start_sales(100, 0));
+
+		// Advance to sale period 1, we should get an AllowedRenewal for task 2001 for the next
+		// sale.
+		advance_to(7);
```

Comment:
Maybe better to use what I introduced here: https://github.com/paritytech/polkadot-sdk/pull/4089/files#diff-9f7f1020b99ae1a246e61572423e1fac95ee6d42802901d822745c74a54488beR213 :D

Reply:
Added in [8ab34b0](https://github.com/paritytech/polkadot-sdk/pull/4099/commits/8ab34b0655c68249ead5ea17f9bb1225b7858658)

Reply:
Ah jeez that resets the reviews now 🤦‍♂️